src/core-services/account/mutations/setUserPermissions.js

```diff
@@ -0,0 +1,80 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+const inputSchema = new SimpleSchema({
+  "accountId": String,
+  "userId": String,
+  "groups": {
+    type: Array, // groupIds that user belongs to
+    optional: true,
+    defaultValue: []
+  },
+  "groups.$": {
+    type: String
+  }
+});
+
+/**
+ * @name accounts/setUserPermissions
+ * @memberof Mutations/Accounts
+ * @summary Adds a group to an account (user) group. This addition to an account  group effectively
+ * adds permissions to account (user)
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {Object} input.groups - groups to append to
+ * @param {String} input.accountId - optional decoded ID of account on which entry should be updated, for admins
+ * @returns {Promise<Object>} with updated account
+ */
+export default async function setUserPermissions(context, input) {
+  const itemsToValidate = { accountId: context.accountId, userId: context.userId, groups: input.groups };
+  inputSchema.validate(itemsToValidate);
+  const { appEvents, collections, userId: userIdFromContext } = context;
+  const { Accounts } = collections;
+  const { accountId } = context;
+  const { groups, shopId } = input;
+
+
+  const account = await Accounts.findOne({ _id: accountId });
+
+  if (!account) throw new ReactionError("not-found", "No account found");
+
+  if (!context.isInternalCall && userIdFromContext !== accountId) {
+    await context.validatePermissions("reaction:accounts", "update", { shopId: account.shopId, legacyRoles: ["reaction-accounts"] });
+  }
+
+  await context.validatePermissions("reaction:accounts", "update", { shopId, legacyRoles: ["admin"] });
+
+  // Update the Reaction Accounts collection with new groups info
+  // This
+  const { value: updatedAccount } = await Accounts.findOneAndUpdate(
+    {
+      _id: accountId
+    },
+    {
+      $addToSet: {
+        groups: {
+          $each: groups
+        }
+      }
+    },
+    {
+      multi: true
+    }
+  );
+
+  if (!updatedAccount) {
+    throw new ReactionError("server-error", "Unable to update account groups");
```

Comment:
Please make this more specific. Put yourself in the shoes of somebody reading this error in the logs. What does it mean? Think about under which conditions the error gets thrown, and make those more apparent in the error message.

Reply:
@rosshadden  ... Actually.. this is hang-over from the original mutation code where this was copied from which used `update`. In any case, I will remove this because the condition here will never be true as we do a check to find out whether the account exists and we throw if it doesnt on line (https://github.com/reactioncommerce/reaction/pull/5835/files#diff-efd372419698e0c60ced0867a59f788eR34)
So when we get here, the condition will always evaluate to false and so we will never throw

Reply:
@rosshadden ... I have updated the err message to  be more specific i.e. this only gets thrown if account is not found